### PR TITLE
Add 2.10.1 Airflow Docker image

### PR DESCRIPTION
**Pull request by michaelfu1029**
*Issue #, if available:*

*Description of changes:*
Adding the folder and code for the Airflow 2.10.1 image. Copied the 2.9.2 image and updated the version and constraints. Built and ran locally with the run.sh script.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mayushko26 — DISMISSED** [review]
Approving as it's a no-impact change, and will be iterated upon in subsequent PR's. However, this change is nearly impossible for a reviewer to confidently review due to the volume of files, so we should give some thought into how we can improve this initial image creation process. Perhaps we could split it up into two PR's: one to copy files into a new x.y.z folder; and a second to make the version-specific changes. This would show the effective diff in the second PR, and draw more attention to the substantive changes made between versions.

Update: The PR now includes two commits - one for copying the 2.9.2 folder into a new 2.10.0 folder, and another for making the version-specific changes. Reviewers can focus on the second commit.

File: images/airflow/2.10.0/python/mwaa/subprocess/subprocess.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def start(
                 exc_info=sys.exc_info(),
             )
 
-    @throttle(seconds=60, instance_level_throttling=True) # avoid excessive calls to process conditions
```
Comment (mayushko26) [inline images/airflow/2.10.0/python/mwaa/subprocess/subprocess.py]: It looks like a few of the recent changes (https://github.com/aws/amazon-mwaa-docker-images/pull/136, https://github.com/aws/amazon-mwaa-docker-images/pull/135) haven't been included here. Probably requires a rebase.

Comment (michaelfu1029) [reply images/airflow/2.10.0/python/mwaa/subprocess/subprocess.py]: Fixed this

**Review by mayushko26 — DISMISSED** [review]

**Review by vishalvijay18 — DISMISSED** [review]

File: images/airflow/2.10.1/bin/airflow-user/post-startup-script-verification [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 echo "Executing post startup script verification"
 
-EXPECTED_AIRFLOW_VERSION="2.10.0"
-EXPECTED_WATCHTOWER_VERSION="3.2.0"
+EXPECTED_AIRFLOW_VERSION="2.10.1"
+EXPECTED_WATCHTOWER_VERSION="3.3.1"
```
Comment (michaelfu1029) [inline images/airflow/2.10.1/bin/airflow-user/post-startup-script-verification:5]: Bumping this as it is changed in the constraints for this version. We will test logging to make sure it doesn't break. Rafid revamped logging for 2.9 so we expect it to go smoothly

**Review by mayushko26 — APPROVED** [review]

**Review by jessefrn1 — APPROVED** [review]

**Review by vishalvijay18 — APPROVED** [review]

**Pull request merged**